dev/reference/sql/data-types/string.md

```diff
@@ -0,0 +1,161 @@
+---
+title: String types
+summary: Learn about the string types supported in TiDB.
+category: reference
+---
+
+# String Types
+
+TiDB supports all the MySQL string types, including `CHAR`, `VARCHAR`, `BINARY`, `VARBINARY`, `BLOB`, `TEXT`, `ENUM`, and `SET`. For more information, [String Types in MySQL](https://dev.mysql.com/doc/refman/5.7/en/string-types.html).
+
+## Supported types
+
+### `CHAR` type
+
+`CHAR` is a fixed length string.  Values stored as `CHAR` are right-padded with spaces to the specified length. M represents the column-length in characters (not bytes).  The range of M is 0 to 255:
+
+```sql
+[NATIONAL] CHAR[(M)] [CHARACTER SET charset_name] [COLLATE collation_name]
+```
+
+### `VARCHAR` type
+
+`VARCHAR` is a string of variable-length. M represents the maximum column length in characters (not bytes). The range of M is 0 to 65,535, but the effective maximum-length will be shorter since the total size of all columns must not exceed 65,535 bytes (the maximum row-size in TIDB).
+
+```sql
+[NATIONAL] VARCHAR(M) [CHARACTER SET charset_name] [COLLATE collation_name]
+```
+
+### `TINYTEXT` type
+
+`TINYTEXT` is a string of variable-length. The length (M) is optional, with the maximum length being restricted to 255 characters:
+
+```sql
+TINYTEXT[(M)] [CHARACTER SET charset_name] [COLLATE collation_name]
+```
+
+### `TEXT` type
+
+`TEXT` is a string of variable-length. M represents the maximum column length ranging from 0 to 65,535. The range of M is 0 to 65,535, but the effective maximum-length will be shorter since the total size of all columns must not exceed 65,535 bytes (the maximum row-size in TIDB):
+
+```sql
+TEXT[(M)] [CHARACTER SET charset_name] [COLLATE collation_name]
+```
+
+### `MEDIUMTEXT` type
+
+`MEDIUMTEXT` is a string of variable-length. M represents the maximum column length ranging from 0 to 16,777,215, but the effective maximum-length will be shorter since the total size of all columns must not exceed 65,535 bytes (the maximum row-size in TIDB):
+
+```sql
+MEDIUMTEXT [CHARACTER SET charset_name] [COLLATE collation_name]
+```
+
+### `LONGTEXT` type
+
+`LONGTEXT` is a string of variable-length. M represents the maximum column length ranging from 0 to 4,294,967,295, but the effective maximum-length will be shorter since the total size of all columns must not exceed 65,535 bytes (the maximum row-size in TIDB):
+
+```sql
+LONGTEXT [CHARACTER SET charset_name] [COLLATE collation_name]
+```
+
+### `BINARY` type
+
+The `BINARY` type is similar to the `CHAR` type, but stores binary byte strings rather than nonbinary character strings. M represents the maximum length in bytes:
+
+```sql
+BINARY(M)
+```
+
+### `VARBINARY` type
+
+The `VARBINARY` type is similar to the `VARCHAR` type, but stores binary byte strings rather than nonbinary character strings. M represents the maximum length, up to 65,535 bytes:
```

Comment:
```suggestion
The `VARBINARY` type is similar to the `VARCHAR` type, but stores binary byte strings rather than nonbinary character strings. M represents the maximum length, ranging from 0 to 65,535 bytes:
```